ui_components/src/ramen/RamenShop.js

```diff
@@ -0,0 +1,231 @@
+                <div className="special_ramen_name">{ramenInfo.label}</div>
+                <div className="special_ramen_description">{ramenInfo.description}</div>
+                <div className="special_ramen_effect">{ramenInfo.effect}</div>
+                <div className="special_ramen_duration">Duration: {ramenInfo.duration} minutes</div>
+                <div className="special_ramen_button" onClick={() => PurchaseSpecialRamen(ramenInfo.ramen_key)}><>&yen;</>{ramenInfo.cost}</div>
+            </div>
+        );
+    }
+    function PurchaseBasicRamen(ramen_key) {
+        apiFetch(
+            ramenShopAPI,
+            {
+                request: 'PurchaseBasicRamen',
+                ramen_key: ramen_key,
+            }
+        ).then((response) => {
+            if (response.errors.length) {
+                handleErrors(response.errors);
+                return;
+            }
+            setPlayerDataState(response.data.player_data);
+            setPlayerResourcesDataState(response.data.player_resources);
+        });
+    }
+    function PurchaseSpecialRamen(ramen_key) {
+        apiFetch(
+            ramenShopAPI,
+            {
+                request: 'PurchaseSpecialRamen',
+                ramen_key: ramen_key,
+            }
+        ).then((response) => {
+            if (response.errors.length) {
+                handleErrors(response.errors);
+                return;
+            }
+            setPlayerDataState(response.data.player_data);
+            setMysteryRamenDetailsState(response.data.mystery_ramen_details);
+            setCharacterRamenDataState(response.data.character_ramen_data);
+            openModal({
+                header: 'Confirmation',
+                text: response.data.response_message,
+                ContentComponent: null,
+                onConfirm: null,
+            });
+        });
+    }
+    function PurchaseMysteryRamen() {
+        apiFetch(
+            ramenShopAPI,
+            {
+                request: 'PurchaseMysteryRamen'
+            }
+        ).then((response) => {
+            if (response.errors.length) {
+                handleErrors(response.errors);
+                return;
+            }
+            setPlayerDataState(response.data.player_data);
+            setMysteryRamenDetailsState(response.data.mystery_ramen_details);
+            setCharacterRamenDataState(response.data.character_ramen_data);
+            openModal({
+                header: 'Confirmation',
+                text: response.data.response_message,
+                ContentComponent: null,
+                onConfirm: null,
+            });
+        });
+    }
+    function handleErrors(errors) {
+        openModal({
+            header: 'Error',
+            text: errors,
+            ContentComponent: null,
+            onConfirm: null,
+        });
+    }
+    return (
+        <div className="ramen_shop_container">
```

Comment:
There's a lot here - Please split the contents into 2-4 components representing semantic sections of the page